jitbuilder/compile/Method.hpp

```diff
@@ -166,7 +167,7 @@ class ResolvedMethod : public ResolvedMethodBase, public Method
                                TR::SymbolReferenceTable *symRefTab);
 
    protected:
-   const char *_fileName;
+   std::string _fileName;
```

Comment:
So, in the OMR core, I agree that we need to be careful with not using the system allocator. However, in JitBuilder, we have to be careful about where we draw the line. 

I have experience the [ `PersistentMemory` destructor assertion](https://github.com/eclipse/omr/blob/7b53b24271a7e10407d7bd4ccedf69fe1964dcc3/compiler/env/TRMemory.hpp#L684) prevent me from wrapping JitBuilder with a library, because in that wrapper I ceded memory management of the JitBuilder objects to the library -- so when it went to destroy the objects, i would fail.